### PR TITLE
Change elevation profile name

### LINKED ISSUE #58455: Use custom title for Elevation Profile window
### Feature description

The Elevation Profile tool really is a major step towards taking 3D data into account.

Sometimes it would be very handy to be able to change the profile window title to have something more meaningful (e.g. a name linked to the path used to generate the profile such as "profile along road XX").

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Simon-Lopez**
## Description

This PR is a simple proposal to answer #58455.
It adds a "Change Progile Name" menu item in the Options sub-menu of the Elevation Profile widget in order to allow the user to change the profile windows name to anything more meaningfull to him/her. The change is made through a modal QInputDialog.

image

image

Fixes #58455

### CODE REVIEW

**Comment by DelazJ** [comment]
Great initiative @Simon-Lopez  👍🏿 . thanks.
There is a dedicated class for dialog renaming you might want to use here: [QgsNewNameDialog](https://github.com/qgis/QGIS/blob/master/src/gui/qgsnewnamedialog.h).
Also, instead of "Change profile name", maybe "Rename profile" or "Rename view" are more inline with QGIS practices.
Questions: should this be the first or the last option in the drop down? In map view settings, it is at the end I think. I was also wondering if that dialog should also popup at the profile creation time, as done for layouts?

**Comment by Simon-Lopez** [comment]
Thanks so much for the feedback @DelazJ. (I'm a newbie here...)
I'm making the changes you propose.
And, yes, it seems a good idea to follow the "map view settings" layout.

**Comment by Simon-Lopez** [comment]
I've applied your suggestions @DelazJ.

File: src/app/elevation/qgselevationprofilewidget.cpp [hunk 1]
```diff
@@ -417,6 +418,12 @@ QgsElevationProfileWidget::QgsElevationProfileWidget( const QString &name )
 
   mOptionsMenu->addAction( mSettingsAction );
 
+  mOptionsMenu->addSeparator();
+
+  mRenameProfileAction = new QAction( tr( "Rename profile" ), this );
```
Comment (nyalldawson) [inline src/app/elevation/qgselevationprofilewidget.cpp:423]:
```suggestion
  mRenameProfileAction = new QAction( tr( "Rename Profile…" ), this );
```

File: src/app/elevation/qgselevationprofilewidget.cpp [hunk 2]
```diff
@@ -962,6 +969,18 @@ void QgsElevationProfileWidget::axisScaleLockToggled( bool active )
   mCanvas->setLockAxisScales( active );
 }
 
+void QgsElevationProfileWidget::renameProfileTriggered()
+{
+  QgsNewNameDialog dlg( tr( "elevation profile" ), canvasName(), {}, {}, Qt::CaseSensitive, this );
+  dlg.setWindowTitle( tr( "Rename elevation profile" ) );
```
Comment (nyalldawson) [inline src/app/elevation/qgselevationprofilewidget.cpp:975]:
```suggestion
  dlg.setWindowTitle( tr( "Rename Elevation Profile" ) );
```

**Comment by nyalldawson** [comment]
Nice work @Simon-Lopez  -- just needed some capitalization fixes to match QGIS HIG guidelines

**Review by nyalldawson — APPROVED** [review]

**Comment by DelazJ** [comment]
Good job @Simon-Lopez  👍🏿

**Pull request merged**